pkg/gameserverallocations/allocation_cache.go

```diff
@@ -275,14 +276,16 @@ func compareGameServers(p *agonesv1.Priority, gs1, gs2 *agonesv1.GameServer) int
 		counter2, ok2 := gs2.Status.Counters[p.Key]
 		// If both game servers have the Counter
 		if ok1 && ok2 {
-			if counter1.Count < counter2.Count {
+			availCapacity1 := counter1.Capacity - counter1.Count
+			availCapacity2 := counter2.Capacity - counter2.Count
+			if availCapacity1 < availCapacity2 {
 				return -1
 			}
-			if counter1.Count > counter2.Count {
+			if availCapacity1 > availCapacity2 {
 				return 1
 			}
-			if counter1.Count == counter2.Count {
-				return 0
+			if availCapacity1 == availCapacity2 {
+				return 0 // TODO: Should we tie-break on Capacity and/or Count?
```

Comment:
Should we tie-break on Capacity and/or Count before returning that two game servers with the same available capacity are equal?

Reply:
I was digging through the implementation, and I think it's fine we don't tie break here. 

We end up tie breaking anyway in 

https://github.com/googleforgames/agones/blob/2f8a583ec6430a093944746e83d68f2a6ed0e6be/pkg/gameserverallocations/allocation_cache.go#L176-L176

So I think it makes sense to keep that tie breaking in a single place.

Reply:
Ah, that's right.